apps/web/modules/event-types/views/event-types-single-view.tsx

```diff
@@ -400,38 +400,47 @@ const EventTypePage = (props: EventTypeSetupProps) => {
 
   useEffect(() => {
     const handleRouteChange = (url: string) => {
+      // If navigating to /event-types then skip this so we can leave without assigning hosts
       const paths = url.split("/");
 
-      // Check if event is managed event type - skip if there is assigned users
-      const assignedUsers = eventType.children;
+      // Use form values as that is what is updated onSuccess - eventType is not inSync
+      const assignedUsers = formMethods.getValues("children");
+      const assignAllTeamMembers = formMethods.getValues("assignAllTeamMembers");
       const isManagedEventType = eventType.schedulingType === SchedulingType.MANAGED;
-      if (eventType.assignAllTeamMembers) {
-        return;
-      } else if (isManagedEventType && assignedUsers.length > 0) {
+
+      // Skip if event is managed event type - skip if there is assigned users or assignAllTeamMembers is true
+      if (isManagedEventType && (assignAllTeamMembers || assignedUsers.length > 0)) {
         return;
       }
 
-      const hosts = eventType.hosts;
-      if (
-        !leaveWithoutAssigningHosts.current &&
-        !!team &&
-        (hosts.length === 0 || assignedUsers.length === 0) &&
-        (url === "/event-types" || paths[1] !== "event-types")
-      ) {
-        setIsOpenAssignmentWarnDialog(true);
-        setPendingRoute(url);
-        router.events.emit(
-          "routeChangeError",
-          new Error(`Aborted route change to ${url} because none was assigned to team event`)
-        );
-        throw "Aborted";
+      const hosts = formMethods.getValues("hosts");
+      // Skip for round-robin
+      const isRoundRobin = eventType.schedulingType === SchedulingType.ROUND_ROBIN;
+      if (isRoundRobin) {
+        if (hosts.length > 0 || assignedUsers.length > 0) {
+          return;
+        }
+      }
+
+      const isInForm = paths[1] === "event-types" && Boolean(paths[2]);
+      if (isInForm || url.includes("skipConfirmation=true")) {
+        return;
       }
```

Comment:
Added a new query param when we leave to the destination route so we can easily know when to avoid calling this handler